go-eicio/writer.go

```diff
@@ -15,11 +17,23 @@ func Create(filename string) (*Writer, error) {
 	if err != nil {
 		return nil, err
 	}
+
+	if strings.HasSuffix(filename, ".gz") {
+		return NewGzipWriter(file)
+	}
+
 	return NewWriter(file), nil
 }
 
-func (wrt *Writer) Close() {
-	wrt.byteWriter.(*os.File).Close()
+func (wrt *Writer) Close() error {
+	flusher, ok := wrt.byteWriter.(Flusher)
```

Comment:
Actually, it looks to me like there may be a bug with the compress/gzip code.  I'm only calling Flush() because Close() did not seem to be flushing first.